### ISSUE TITLE
Rotating AES keys

### ISSUE THREAD

**Issue by OdedRaiches** [pkasila/gorm-crypto#15]
**Describe the bug**
Can't seem to successfully rotate AES keys with InitWithFallbacks().
I want to be able to replace AES encryption key and still have the old ones as fallback (I understand this is the feature capability).

**To Reproduce**
Code to reproduce the behavior:
```
func InitGormCrypt(aesPasswords []string) error {
	var algs []algorithms.Algorithm
	for _, aesPass := range aesPasswords {
		aesAlg, err := algorithms.NewAES256CBC([]byte(aesPass))
		if err != nil {
			return err
		}

		// NOTE: when a rotation of an encryption key or encryption method is required,
		// 		 define that encryption algorithm and APPEND it to the algs slice
		// Example found in Fallbacks: https://github.com/pkasila/gorm-crypto#fallbacks
		algs = append(algs, aesAlg)
	}

	gormcrypto.InitWithFallbacks(algs, []serialization.Serializer{serialization.NewJSON()})

	return nil
}
...
...
func TestPostgres(t *testing.T) {
	...
	oldPassword := "passphrasewhichneedstobe32bytes!"
	// one char difference
	newPassword := "passphrasewhichneedstobe32cytes!"

	err := InitGormCrypt([]string{newPassword, oldPassword})
	...
}
```

**Expected behavior**
I expect already encrypted fields to be scanned correctly, instead I get this:
```
failed to get object, err: sql: Scan error on column index 1, name "password": invalid character 'ֳ˜' looking for beginning of value
```

In addition, I noticed that it works only when putting just one algorithm in the slice (that has the original encryption key). 
Even doing (same twice):
```
InitGormCrypt([]string{oldPassword , oldPassword })
```

It still throws
```
failed to get object, err: sql: Scan error on column index 1, name "password": invalid character 'P' looking for beginning of value
```

**Comment by OdedRaiches** [comment]
@pkasila

**Comment by pkasila** [comment]
Hello, I've reproduced this bug and I'll investigate it, seems like there were some breaking changes in these several months, so it doesn't work properly with the latest version of Go and GORM.

**Comment by pkasila** [comment]
@OdedRaiches I've tried to fix this issue in e34c7d0, I've also added some tests for multiple AES keys. Can you check whether this commit actually fixes it?

**Comment by pkasila** [comment]
Fixed in [`v1.0.3`](https://github.com/pkasila/gorm-crypto/releases/tag/v1.0.3)
